modules/custom-status/custom-status.php

```diff
@@ -1314,5 +1347,17 @@ public function fix_post_row_actions( $actions, $post ) {
 		$actions['view'] = '<a href="' . esc_url( $preview_link ) . '" title="' . esc_attr( sprintf( __( 'Preview &#8220;%s&#8221;' ), $post->post_title ) ) . '" rel="permalink">' . __( 'Preview' ) . '</a>';
 		return $actions;
 	}
+
+	public function add_custom_status_ui_properties( $custom_status ) {
+		// Add 'required_user_logins' data with display names for status restrictions UI
+		$required_user_ids = $custom_status->required_user_ids ?? [];
+
+		$custom_status->required_user_logins = array_filter(array_map( function ( $user_id ) {
+			$user = get_user_by( 'ID', $user_id );
+			return $user instanceof WP_User ? $user->user_login : false;
```

Comment:
I do like the safety check here. I'd wanna do more, just to be safe:

- Hook into `do_action( 'deleted_user', $id, $reassign, $user );` to ensure that the user IDs in the term_meta is always correct.
- In the event that a user is not available or is not the right type, that ID should immediately be dropped.